cpp/autosar/src/rules/M5-14-1/RightHandOperandOfALogicalAndOperatorsContainSideEffects.ql

```diff
@@ -19,10 +19,44 @@ import codingstandards.cpp.autosar
 import codingstandards.cpp.SideEffect
 import codingstandards.cpp.sideeffect.DefaultEffects
 
+/**
+ * an operator that does not evaluate its operand
+ * `decltype` also has a non evaluated operand but cannot be used in a `BinaryLogicalOperation`
+ */
+class UnevaluatedOperand extends Expr {
```

Comment:
There is an opportunity for reuse using [isUnevaluated](https://codeql.github.com/codeql-standard-libraries/cpp/semmle/code/cpp/exprs/Expr.qll/predicate.Expr$Expr$isUnevaluated.0.html). 

Perhaps we can create an extended predicate in our `Expr` module that adds the `decval` case in addition to the cases covered by `isUnevaluated`.

Reply:
great catch! cant believe I missed this, oh I actually think that case is not relevant now, cannot create a testcase that would use that in the conditional (think its not possible)